x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/RollupShardIndexer.java

```diff
@@ -218,24 +229,22 @@ public void afterBulk(long executionId, BulkRequest request, Throwable failure)
             }
         };
 
-        return BulkProcessor.builder(client::bulk, listener, "rollup-shard-indexer")
+        return BulkProcessor2.builder(client::bulk, listener, client.threadPool())
             .setBulkActions(ROLLUP_BULK_ACTIONS)
             .setBulkSize(ROLLUP_BULK_SIZE)
-            // execute the bulk request on the same thread
-            .setConcurrentRequests(0)
```

Comment:
I think this is a change in runtime behaviour? `BulkProcessor2` would execute the bulk request on different threads and if multiple bulk requests exceed max bytes in flight, then a rejected exception is thrown. Just double checking, I think `BulkProcessor2` would work well in this context.